test/sanity/code-smell/update-bundled.py

```diff
@@ -0,0 +1,104 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+import glob
+import json
+import os.path
+import sys
+from distutils.version import LooseVersion
+
+import packaging.specifiers
+
+from ansible.module_utils.urls import open_url
+
+
+def compat_libs(basedir):
+    for filename in glob.glob(os.path.join(basedir, 'lib/ansible/compat/*/__init__.py')):
+        if 'compat/tests' in filename:
+            # compat/tests doesn't bundle any code
+            continue
+        yield filename
+
+    yield os.path.join(basedir, 'lib/ansible/module_utils/distro/__init__.py')
+    yield os.path.join(basedir, 'lib/ansible/module_utils/six/__init__.py')
+    yield os.path.join(basedir, 'lib/ansible/module_utils/compat/ipaddress.py')
+    # backports.ssl_match_hostname should be moved to its own file
+    yield os.path.join(basedir, 'lib/ansible/module_utils/urls.py')
+
+
+def get_bundled_metadata(filename):
+    with open(filename, 'r') as module:
+        for line in module:
+            if line.strip().startswith('_BUNDLED_METADATA'):
+                data = line[line.index('{'):].strip()
+                break
+        else:
+            raise Exception('{0} contained no metadata.  Could not check for updates'.format(filename))
+        metadata = json.loads(data)
+    return metadata
+
+
+def get_latest_applicable_version(pypi_data, constraints=None):
+    latest_version = "0"
+    if 'version_constraints' in metadata:
+        version_specification = packaging.specifiers.SpecifierSet(metadata['version_constraints'])
+        for version in pypi_data['releases']:
+            if version in version_specification:
+                if LooseVersion(version) > LooseVersion(latest_version):
+                    latest_version = version
+    else:
+        latest_version = pypi_data['info']['version']
+
+    return latest_version
+
+
+if __name__ == '__main__':
+    needs_attention = False
+    basedir = os.path.join(os.path.dirname(__file__), '../../..')
+
+    for filename in compat_libs(basedir):
+        filename = os.path.normpath(filename)
+        try:
+            metadata = get_bundled_metadata(filename)
+        except Exception as e:
+            needs_attention = True
+            print('WARNING: {0}'.format(e))
+            continue
+
+        pypi_fh = open_url('https://pypi.org/pypi/{0}/json'.format(metadata['pypi_name']))
+        pypi_data = json.loads(pypi_fh.read().decode('utf-8'))
+
+        constraints = metadata.get('version_constraints', None)
+        latest_version = get_latest_applicable_version(pypi_data, constraints)
+
+        if LooseVersion(metadata['version']) < LooseVersion(latest_version):
+            needs_attention = True
+            print('UPDATE: {0} from {1} to {2} {3}'.format(
```

Comment:
Messages should be in the format: `path: message`

Also, include a `update-bundled.json` file with:

```json
{
    "always": true,
    "output": "path-message"
}
```

That will tell ansible-test when to run the test and what format to expect messages in.

Reply:
Done.